src/shared/json.h

```diff
@@ -259,6 +263,59 @@ enum {
 #define JSON_BUILD_HEX(p, n) _JSON_BUILD_HEX, (const void*) { p }, (size_t) { n }
 #define JSON_BUILD_ID128(id) _JSON_BUILD_ID128, (const sd_id128_t*) { &(id) }
 #define JSON_BUILD_BYTE_ARRAY(v, n) _JSON_BUILD_BYTE_ARRAY, (const void*) { v }, (size_t) { n }
+#define JSON_BUILD_IN4_ADDR(v) JSON_BUILD_BYTE_ARRAY(v, sizeof(struct in_addr))
+#define JSON_BUILD_IN6_ADDR(v) JSON_BUILD_BYTE_ARRAY(v, sizeof(struct in6_addr))
+#define JSON_BUILD_IN_ADDR(v, f) JSON_BUILD_BYTE_ARRAY((v)->bytes, FAMILY_ADDRESS_SIZE_SAFE(f))
+#define JSON_BUILD_ETHER_ADDR(v) JSON_BUILD_BYTE_ARRAY((v)->ether_addr_octet, sizeof(struct ether_addr))
+#define JSON_BUILD_HW_ADDR(v) JSON_BUILD_BYTE_ARRAY((v)->bytes, (v)->length) /* This evaluates the argument twice. */
```

Comment:
hmm, maybe define a new enum _JSON_BUILD_HW_ADDR for this one and pass v directly, so that json_build() will receibe the structure as a whole and we avoid the double eval